### ISSUE TITLE
Mounted route fails if it has a dot in path (at first part)

### ISSUE THREAD

**Issue by gencer** [rails/rails#36491]
### Steps to reproduce
1. Create a rails project.
2. Add `grape` as a dependency and install
3. Open `routes.rb` and add following:

```
# frozen_string_literal: true

class DomainConstraint
  def initialize(domain)
    @domain = domain.downcase
  end

  def matches?(request)
    Rails.logger.debug "Request for host: #{request.host}"
    @domain == request.host.downcase
  end
end

Rails.application.routes.draw do
  constraints DomainConstraint.new(domain) do
    mount API => '/' # this is grape API mounted in Rails via '/' path.
  end
end
```

### Expected behavior

```http
GET http://example.com/gencer/test
GET http://example.com/.gencer/test
```

Those two request should be passed to grape because I've mounted base path to Grape as '/'.

### Actual behavior
Only 

```http
GET http://example.com/gencer/test
```

**successfully** passed to grape.

```http
GET http://example.com/.gencer/test
```

throws **404** not found error.

```
No route matches [GET] "/.gencer/test"
```

Related Grape code to capture path:

```
get ':path', requirements: { path: /.*/ } do
   ...
end
```

Even constraint is not executed if I use dot `.`. I've added a logging to `matches?` method. If I use dot in path, log is not present.

The thing is, This was working in the past. Now it isn't. Something has changed but whats changed, that I don't know for sure.

### System configuration
**Rails version**: 5.2.3

**Ruby version**: 2.6.3

## Demo

Clone https://github.com/gencer/mount_test

run `bundle install && rails s`

Go to

```http
GET http://localhost:3000/gencer/demo
```

It will show you

```json
{ "success": true }
```

Now go to:

```http
GET http://localhost:3000/.gencer/demo
```

You will get 404 / No route matched.

Refer: `config/routes.rb`

## Update

This only happens if base path has a dot in it. For example this will success:

```http
GET http://localhost:3000/demo/.gencer/demo
```

```json
{ "success": true }
```

So, If path start with dot (first folder) than route does not match with any.

## Update 2:

I tested against Rails `6.0.0.rc1` and dotted paths working fine there. However, for me, upgrading to Rails 6 is not possible at the moment because many of my gems depends on `rails < 6`

**Comment by gencer** [comment]
## Demo

Clone https://github.com/gencer/mount_test

run `rails s`

Go to

```http
GET http://localhost:3000/gencer/demo
```

It will show you

```json
{ "success": true ]
```

Now go to:

```http
GET http://localhost:3000/.gencer/demo
```

You will get 404 / No route matched.

**Comment by gencer** [comment]
This has been fixed in `rails-6.0`. Closing.
